src/plugin/file/kml/kmlsource.js

```diff
@@ -432,6 +432,27 @@ plugin.file.kml.KMLSource.prototype.onImportComplete = function(opt_event) {
 };
 
 
+/**
+ * @inheritDoc
+ */
+plugin.file.kml.KMLSource.prototype.clearQueue = function() {
+  var toRemove = [];
+  for (var key in this.nodeMap_) {
+    var node = this.nodeMap_[key];
+    if (node && node.isDisposed()) {
+      toRemove.push(key);
+    }
+  }
+
+  var nmap = this.nodeMap_;
+  toRemove.forEach(function(key) {
+    nmap[key] = undefined;
```

Comment:
Can this be done directly in the above loop to avoid the extra `toRemove` array?